config/src/main/java/org/springframework/security/config/annotation/web/configurers/SessionManagementConfigurer.java

```diff
@@ -588,6 +594,17 @@ private SessionAuthenticationStrategy getSessionAuthenticationStrategy(H http) {
 		return this.sessionAuthenticationStrategy;
 	}
 
+	private SessionLimitStrategy getSessionLimitStrategy() {
```

Comment:
We can simplify some of this conditional logic by treating `maxSessions` as a `sessionLimitStrategy` internally.

For example `maximumSessions(Integer)`'s implementation could become:

```java
this.sessionLimitStrategy = (authentication) -> maximumSessions;
```

In that case, `this.maximumSessions` would go away.

This is also valuable because it aligns with the common pattern in Spring Security to allow applications to easily override default behavior.